src/lexer/token.h

```diff
@@ -97,7 +97,9 @@ typedef enum {
 #define TOKEN_LENGTH 256
 
 typedef struct {
-    TokenType type;              // What type of token this is.
-    char contents[TOKEN_LENGTH]; // The actual contents of the token.
-    unsigned length;             // How long the token is.
+    TokenType type;                 // What type of token this is.
+    char contents[TOKEN_LENGTH];    // The actual contents of the token.
+    unsigned length;                // How long the token is.
+    char source_file[TOKEN_LENGTH]; // The source file the token was in.
+    long position_in_file;           // Where in the file the token started, as an index.
```

Comment:
Is there a way to get row/column info from this? If not, we should maybe switch to that so we can give helpful error messages.

Reply:
my plan was to have the error reporting find that. It is simpler (i think?) to just crawl to the relevant position and hold onto a few rows of buffer. plus, errors are only usually called a few times, but this is tracked for every token

Reply:
ok so line and column information is now being tracked instead